### PR TITLE
Remove `changelog_url` from `ObjectView` Context

### PR DESCRIPTION
**Pull request by bryanculver**
# Closes: #2885
# What's Changed

Removed deprecated code.

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [NA] Attached Screenshots, Payload Example
- [NA] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [NA] Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md [hunk 1]
```diff
@@ -130,6 +130,10 @@ These endpoints `/ipam/roles/`, `/dcim/rack-roles/` and `/dcim/device-roles/` ar
 
 ## UI, GraphQL, and REST API Filter Changes
 
+### Removed Changelog URL from View Context
+
+`changelog_url` is no longer provided in the `ObjectView` context. To get a models' changelog URL, you can retrieve it from the model itself if it supports it: `model_instance.get_changelog_url()`.
```
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:135]:
```suggestion
`changelog_url` is no longer provided in the `ObjectView` context. To get a model instance's changelog URL, you can retrieve it from the instance itself if it supports it: `model_instance.get_changelog_url()`.
```

**Review by glennmatthews — APPROVED** [review]

File: nautobot/core/views/generic.py [hunk 2]
```diff
@@ -89,52 +86,19 @@ def get_extra_context(self, request, instance):
...
     def get(self, request, *args, **kwargs):
         """
         Generic GET handler for accessing an object by PK or slug
         """
         instance = get_object_or_404(self.queryset, **kwargs)
 
-        changelog_url = None
-
-        if isinstance(instance, ChangeLoggedModel):
-            changelog_url = instance.get_changelog_url()
```
Comment (glennmatthews) [inline nautobot/core/views/generic.py:128]: Looks like there's similar code in `nautobot/core/views/renderers.py` as well.

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md [hunk 3]
```diff
@@ -130,6 +130,10 @@ These endpoints `/ipam/roles/`, `/dcim/rack-roles/` and `/dcim/device-roles/` ar
 
 ## UI, GraphQL, and REST API Filter Changes
 
+### Removed Changelog URL from View Context
+
+`changelog_url` is no longer provided in the `ObjectView` context. To get a model instance's changelog URL, you can retrieve it from the instance itself if it supports it: `model_instance.get_changelog_url()`.
```
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:135]: There's a reference to `changelog_url` in `nautobot/docs/plugins/development.md` that probably needs to be revised.

**Review by glennmatthews — CHANGES_REQUESTED** [review]

File: nautobot/core/views/generic.py (hunk 2 above)
Comment (bryanculver) [reply nautobot/core/views/generic.py:128]: Shoulda ran the grep... 🤦

File: nautobot/docs/plugins/development.md [hunk 4]
```diff
@@ -1090,7 +1090,6 @@ Other context keys may be available for certain views:
 
 You may see other context keys as well, but any not documented above should not be relied upon as they may be removed in a future release. Some examples of those are:
 
-* `changelog_url`: This can now be retrieved from the object itself, via `object.get_changelog_url`, if the object supports change-logging
```
Comment (glennmatthews) [inline nautobot/docs/plugins/development.md:1093]: Possibly add a version-changed admonition  below this list, something like:

```markdown
+/- 2.0.0
    The `changelog_url` context key was removed. Use `object.get_changelog_url` instead.
```

Comment (bryanculver) [reply nautobot/docs/plugins/development.md:1093]: What about `---` since it was removed?

**Pull request merged**